test/controllers/StatusControllerSpec.scala

```diff
@@ -805,14 +811,114 @@ class StatusControllerSpec extends AmlsSpec with PaymentGenerator with PrivateMe
+
+        result returns 1
+      }
+
+      "return 0 if there is amls ref number available and 1 read notification" in new Fixture with FutureAssertions {
+
+        when(controller.notificationConnector.fetchAllByAmlsRegNo(eqTo(amlsRegistrationNumber), any())(any(), any()))
+          .thenReturn(Future.successful(Seq(NotificationRow(None, None, None, false, DateTime.now(), true, amlsRegistrationNumber, "", IDType("")))))
+
+        val result = controller.countUnreadNotifications(Some(amlsRegistrationNumber), None, ("", ""))
+
+        result returns 0
+      }
+
+      "return 0 if there is safe id available but no notifications returned" in new Fixture with FutureAssertions {
+
+        when(controller.notificationConnector.fetchAllBySafeId(eqTo("internalId"), any())(any(), any()))
+          .thenReturn(Future.successful(Seq()))
+
+        val result = controller.countUnreadNotifications(None, Some("internalId"), ("", ""))
+
+        result returns 0
+      }
+
+      "return 1 if there is safe id available and 1 unread notification" in new Fixture with FutureAssertions {
+
+        when(controller.notificationConnector.fetchAllBySafeId(eqTo("internalId"), any())(any(), any()))
+          .thenReturn(Future.successful(Seq(NotificationRow(None, None, None, false, DateTime.now(), false, amlsRegistrationNumber, "", IDType("")))))
+
+        val result = controller.countUnreadNotifications(None, Some("internalId"), ("", ""))
+
+        result returns 1
+      }
+
+      "return 1 if there is safe id available and 1 unread notification and 1 read notification" in new Fixture with FutureAssertions {
+
+        when(controller.notificationConnector.fetchAllBySafeId(eqTo("internalId"), any())(any(), any()))
+          .thenReturn(Future.successful(Seq(NotificationRow(None, None, None, false, DateTime.now(), false, amlsRegistrationNumber, "", IDType("")),
+            NotificationRow(None, None, None, false, DateTime.now(), true, amlsRegistrationNumber, "", IDType("")))))
+
+        val result = controller.countUnreadNotifications(None, Some("internalId"), ("", ""))
+
+        result returns 1
+      }
+
+      "return 0 if there is safe id available and 1 read notification" in new Fixture with FutureAssertions {
+
+        when(controller.notificationConnector.fetchAllBySafeId(eqTo("internalId"), any())(any(), any()))
+          .thenReturn(Future.successful(Seq(NotificationRow(None, None, None, false, DateTime.now(), true, amlsRegistrationNumber, "", IDType("")))))
+
+        val result = controller.countUnreadNotifications(None, Some("internalId"), ("", ""))
+
+        result returns 0
+      }
+
+      "throw an exception if no match found" in new Fixture with FutureAssertions {
+
+        val result = intercept[MatchError] {
+          controller.countUnreadNotifications(null, null, ("", ""))
+        }
+
+        result.getMessage() must include("Could not match amls ref number or safe id against given conditions.")
```

Comment:
I think it might be safer to log a warn and return zero. It would then be non blocking for the user for the sake of a message count.

Reply:
Done :)